### PR TITLE
refactor: remove a lot of unused code

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- refactor: move schema next to script that uses it
- fix: add tests as knip entry files
- chore: update dependencies (and faker -> faker-js)
- refactor: remove unused actions
- refactor: remove unused exports and variables
- refactor: remove more unused exports
- refactor: remove unused selectors
- refactor: remove unused variables
- refactor: remove unused action
- refactor: remove unused prop-types
- refactor: remove more unused exports

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This is most, but not all, of the unused stuff that knip found.  The rest are a little trickier to remove and this seemed like enough for one PR.

### CODE REVIEW

**Review by raisedadead — CHANGES_REQUESTED** [review]
If you are to be moving files, please use kebab-case `tools/scripts/build/schema/trendingSchema.ts`

**Review by raisedadead — APPROVED** [review]

File: client/package.json [hunk 1]
```diff
@@ -133,13 +135,16 @@
...
+    "@faker-js/faker": "7.6.0",
     "@testing-library/jest-dom": "5.16.5",
     "@testing-library/react": "12.1.5",
+    "@types/faker": "6.6.9",
     "autoprefixer": "10.4.13",
     "babel-plugin-transform-imports": "2.0.0",
     "chokidar": "3.5.3",
     "copy-webpack-plugin": "9.1.0",
     "core-js": "2.6.12",
+    "faker": "6.6.6",
```
Comment (naomi-lgbt) [inline client/package.json:147]: We're using `@faker-js/faker` - are we sure we need `faker` too?

Additionally, `@faker-js/faker` uses TypeScript so we shouldn't need `@types/faker`?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: client/package.json (hunk 1 above)
Comment (raisedadead) [reply client/package.json:147]: _Can we fake using faker so that we use faker by faking it?_

Comment (ojeytonwilliams) [reply client/package.json]: @naomi-lgbt nah, that was just a mistake.  I forgot that the original faker repo had been replaced with nonsense and I somehow forgot to remove it :face_exhaling:

Comment (ahmadabdolsaheb) [reply client/package.json]: I am removing faker from the client in the Analytics migration pr, so we could leave it off here.

**Review by naomi-lgbt — APPROVED** [review]

File: client/package.json (hunk 1 above)
Comment (naomi-lgbt) [reply client/package.json]: Is this okay to merge or are we waiting for the analytics pr?

**Review by raisedadead — APPROVED** [review]

**Pull request merged**